modules/optimization/src/base/OptimizationApp.C

```diff
@@ -43,6 +44,7 @@ OptimizationApp::registerAll(Factory & f, ActionFactory & af, Syntax & syntax)
   registerSyntaxTask(
       "AddOptimizationReporterAction", "OptimizationReporter", "add_optimization_reporter");
   registerMooseObjectTask("add_optimization_reporter", OptimizationReporterBase, false);
+  appendMooseObjectTask("add_optimization_reporter", Reporter);
```

Comment:
What's up with this change?

Reply:
For OptimizationReporter to be able to create the OptimizationDataHelper this
needs to be included.